marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -151,11 +151,28 @@ protected JsonNode determineIngestionEndpointParams(Map<String, String> params)
         }
 
         if (!endpointParams.hasNonNull("workUnit")) {
-            ObjectNode defaultWorkUnit = objectMapper.createObjectNode();
-            defaultWorkUnit.put("uriprefix", params.get("uriprefix")!=null ? params.get("uriprefix"):"");
+            defaultWorkUnit = objectMapper.createObjectNode();
+            defaultWorkUnit.put("uriprefix", params.get("uriprefix") != null ? params.get("uriprefix") : "");
+            buildWorkUnitFromParams(params);
             endpointParams.set("workUnit", defaultWorkUnit);
         }
 
         return endpointParams;
     }
+
+    protected void buildWorkUnitFromParams(Map<String, String> params) {
+        if (params.get("collections") != null) {
```

Comment:
A more succinct approach with Jackson is to use "has":

    if (params.has("collections"))

Also, all 4 conditionals are the same, so you could simplify this like so:

    Stream.of("collections", "permissions", "sourcename", "sourcetype", "uriprefix").forEach(key -> {
      if (params.containsKey(key)) {
        defaultWorkUnit.put(key, params.get(key));
      }
    }

The one downside is we have to use the lower-casing that Spark requires. I think we can live with that for now though.

Reply:
Updated the changes